src/query/tsdb/remote/compressed_codecs.go

```diff
@@ -434,3 +444,90 @@ func DecodeCompressedFetchResponse(
 		nil,
 	), nil
 }
+
+func filterTagIterators(
+	filterTagNames [][]byte,
+	iter ident.TagIterator,
+) (ident.TagIterator, error) {
+	tags := ident.NewTags()
+	for iter.Next() {
+		tag := iter.Current()
+		tagName := tag.Name.Bytes()
+		for _, filter := range filterTagNames {
+			if bytes.Equal(filter, tagName) {
+				tags.Append(tag)
+			}
+		}
+	}
+
+	if err := iter.Err(); err != nil {
```

Comment:
Hm do we also close the iterator from somewhere?  Perhaps we need to `iter.Close()` after checking for `iter.Err()`?

Reply:
Won't the underlying data be released in the cleanup() called in server.Search? If we close here we lose the data in `tags` don't we?